plugins/modules/discovery_intent.py

```diff
@@ -490,6 +601,164 @@ def get_creds_ids_list(self):
         self.log("Credential Ids list passed is {0}".format(str(self.creds_ids_list)), "INFO")
         return self.creds_ids_list
 
+    def handle_global_credentials(self, response=None):
+        """
+        Method to convert values for create_params API when global paramters
+        are passed as input.
+
+        Parameters:
+            - response: The response collected from the get_all_global_credentials_v2 API
+
+        Returns:
+            - global_cred_dict  : The dictionary containing list of IDs of various types of
+                                    Global credentials.
+        """
+
+        global_credentials = self.validated_config[0].get("global_credentials")
+        cli_credentials_list = global_credentials.get('cli_credentials_list')
+        http_read_credential_list = global_credentials.get('http_read_credential_list')
+        http_write_credential_list = global_credentials.get('http_write_credential_list')
+        snmp_v2_read_credential_list = global_credentials.get('snmp_v2_read_credential_list')
+        snmp_v2_write_credential_list = global_credentials.get('snmp_v2_write_credential_list')
+        snmp_v3_credential_list = global_credentials.get('snmp_v3_credential_list')
+        net_conf_port_list = global_credentials.get('net_conf_port_list')
+        global_cred_dict = {}
+        if cli_credentials_list:
+            if not isinstance(cli_credentials_list, list):
+                msg = "Global ClI credentials must be passed as a list"
+                self.discovery_specific_cred_failure(msg=msg)
+            if len(cli_credentials_list) > 0:
+                global_cred_dict["cliCredential"] = []
+                cred_len = len(cli_credentials_list)
```

Comment:
we can take this first before checking the len with 0. We are unnecessarily comparing two times.
```
cli_cred_len = len(cli_credentials_list)

If cli_cred_len <= 0:
    Error

if cli_cred_len > 5:
   cli_cred_len = 5
```

Reply:
It can't be less than 0 and for equal to 0 it will be skipped